include/numerics/wrapped_petsc.h

```diff
@@ -0,0 +1,228 @@
+// The libMesh Finite Element Library.
+// Copyright (C) 2002-2020 Benjamin S. Kirk, John W. Peterson, Roy H. Stogner
+
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+#ifndef LIBMESH_WRAPPED_PETSC_H
+#define LIBMESH_WRAPPED_PETSC_H
+
+#include "libmesh/libmesh_config.h"
+
+#ifdef LIBMESH_HAVE_PETSC
+
+// libMesh includes
+#include "libmesh/petsc_solver_exception.h"
+#include "libmesh/petsc_macro.h"
+
+// PETSc includes
+#ifdef I
+# define LIBMESH_SAW_I
+#endif
+
+#include <petscksp.h>
+#include <petscvec.h>
+#include <petscis.h>
+#include <petscmat.h>
+#include <petscviewer.h>
+#include <petscpc.h>
+#include <petscsnes.h>
+#include <petscdm.h>
+#include <petscsf.h>
```

Comment:
I'd like to get rid of this lump-of-dependencies, if possible (or at least replace it with forward declarations), so each WrappedPetsc<Foo> will still only need petscfoo.h.  Any way to move them (and the destroy overrides) to a wrapped_petsc.C?

Reply:
Yeah I wasn't happy with this either. And there was only an issue on the `make installcheck` stage of the build, when the new header is tested without any other surrounding headers. I guess that could actually be fixed by moving the destroy specializations to a .C file... I'll try it.

Reply:
It worked in my local testing, thanks for the suggestion.